sound/soc/sof/intel/hda-loader-skl.c

```diff
@@ -474,7 +476,7 @@ int hda_dsp_cl_boot_firmware_skl(struct snd_sof_dev *sdev)
 				snd_sof_dsp_read(sdev, HDA_DSP_BAR,
 						 HDA_ADSP_FW_STATUS_SKL));
 			dev_err(sdev->dev, "error: Core En/ROM load fail:%d\n", ret);
-			goto irq_err;
+			return ret;
```

Comment:
why is this changed?

Reply:
@plbossart This is a bug, using goto will dump and call powerdown two times. so if fail, just return.

Reply:
@zhuyingjiang sorry, I don't see what would cause this loop to be executed twice. the flow is quite unclear to me here.